src/uu/tail/src/args.rs

```diff
@@ -358,8 +408,32 @@ fn parse_num(src: &str) -> Result<Signum, ParseSizeError> {
 }
 
 pub fn parse_args(args: impl uucore::Args) -> UResult<Settings> {
-    let matches = uu_app().try_get_matches_from(arg_iterate(args)?)?;
-    Settings::from(&matches)
+    let args_vec: Vec<OsString> = args.collect();
+    let clap_args = uu_app().try_get_matches_from(args_vec.clone());
+    let clap_result = match clap_args {
+        Ok(matches) => Ok(Settings::from(&matches)?),
+        Err(err) => Err(err.into()),
+    };
+
+    // clap misinterprets obsolete args as first input (starting with +) or Err
+    if let Ok(settings) = clap_result.as_ref() {
+        // Settings::from() ensures we always have at least one input (STDIN as fallback)
+        let input = settings.inputs.get(0).unwrap();
```

Comment:
Instead of `.get(0).unwrap()`, you can just use `settings.inputs[0]`.

Reply:
Also, I think the best value to check here is `args_vec.get(1)`, because the first input is not necessarily guaranteed to be the first argument. (Though I can't think of a case where your current implementation fails, it's just a bit safer)